### DISCUSSION TITLE
Xiaomi Mi Box Beta Testing Group

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#55] [Announcements]
# Upcoming Xiaomi Mi Box Compatibility
![image](https://user-images.githubusercontent.com/25486599/227838334-031a132f-2ef6-4ec6-8749-f450d18a4d9f.png) ![image](https://user-images.githubusercontent.com/25486599/227838967-f2c0c956-8a46-4951-9ebe-fa9a9dc173e0.png)

This is a beta release for testing only - not production ready

<hr /><em>This discussion was created from the release <a href='https://github.com/PRProd/HA-Firemote/releases/tag/v2.1.0-beta1'>Xiaomi Mi Box Beta 1</a>.</em>

**Comment by PRProd** [comment]
Do you own a Xiaomi Mi Box?  Do you have a moment to test the latest pre-release?  [Version 2.1.0-beta1](https://github.com/PRProd/HA-Firemote/releases/tag/v2.1.0-beta1) is now available for download and testing.  Let me know what you find!  Thanks!

**Comment by moh-Brent** [comment]
@PRProd awesome work on this and soo quickly THANK YOU!!
I have managed to do some testing and these are my findings so far.

**Missing Apps**
TVNZ+
Three Now

**What's not working**
9 dot app button doesn't work
menu button is highlighted orange as if it is the power button - sometimes
- maybe highlight when in the main menu?

**What works + Enhancement**
YT
Netflix
Disney+
plex
Prime Video
YT Kids
Function: Switch Apps
Function: Mute
Function: Reboot
- reboot without an OK or Cancel?

Vol + & -

**Reply by PRProd** [reply]
THANK YOU FOR YOUR TESTING!!

I'll try to address all of your questions

> **Missing Apps**
> TVNZ+
> Three Now

I had left these off because I'm unable to test them.  After I publish beta version 2 (not ready yet), I'll need your help to confirm that they are there and they are working as expected.


> **What's not working**
> 9 dot app button doesn't work

That's unexpected.  hmmm.... we will need to try a few things:
 - Confirm that you have all of your settings properly configured for the Xiaomi Mi Box like this: 
![image](https://user-images.githubusercontent.com/25486599/228049391-5e400c44-7ce7-4337-a767-1566f03802be.png)
 - If those are all set like you see in the image, try creating a brand new card using all of those settings, then check to see if the button works.
 - If the button STILL doesn't function, press the 9 dots app button on your physical remote control to open the Apps menu.  When the apps menu is open, [follow the instructions here](https://github.com/PRProd/HA-Firemote/wiki/How-Can-I-Help%3F#details-for-app-requests) as if you were making an app request, and post the results into this conversation.


> menu button is highlighted orange as if it is the power button - sometimes
> 
> * maybe highlight when in the main menu?

The home button is lit whenever you are at one of the device's "home" screens.  This is true across all of the supported device types and remote styles.  


> Function: Reboot
> 
> * reboot without an OK or Cancel?

What!?  You don't want to accidentally bump your phone and have your device reboot?  :rofl:  I'll see what I can do about adding in a confirmation box so that clumsy people like us can have a little extra safety.


BTW: I'm still very curious, is your device a Mi Box S, or a different Xiaomi model?

**Reply by moh-Brent** [reply]
NOTE: using Beta 2

It is a Mi Box S
APP BUTTON: Tried creating a new Card and that didn't work, the ADB command gave me the below
```
adb_response: >-
  mSurface=Surface(name=com.google.android.tvlauncher/com.google.android.tvlauncher.MainActivity)/@0x976c19f
        mSurface=Surface(name=com.android.systemui.ImageWallpaper)/@0x4e53005
    mCurrentFocus=Window{6bcca04 u0 com.google.android.tvlauncher/com.google.android.tvlauncher.MainActivity}
    mFocusedApp=AppWindowToken{74e6b2e token=Token{30929a9 ActivityRecord{1743930 u0 com.google.android.tvlauncher/.MainActivity t634}}}
```
TVNZ+ and Three Now are only in Firemote options, not Xiaomi and nVidea :)

**Reply by PRProd** [reply]
I don't think you have version 2.  Those apps are definitely available in v2

**Reply by moh-Brent** [reply]
![image](https://user-images.githubusercontent.com/44347375/228112171-4ae0748b-6f29-4556-be7a-68898f441f0c.png)
i'll try reinstalling it

**Reply by moh-Brent** [reply]
OK so redownloading it did the trick, had an error about too many Github requests or something

**Reply by PRProd** [reply]
@moh-Brent - I'm anxious to hear about your testing results after getting the beta2 version successfully installed.  Hopefully your results are a little different?

**Reply by moh-Brent** [reply]
@PRProd he, soo sorry, thought I had replied last night, so the missing Apps are there now :)  the Apps button did not work, but give me some time this avo after work and i'll give it a go without the wife complaining of my playing with stuff :)
FYI, I also tried it on the older Mi Box 3 but it runs like a dog, almost non usable, will give that a good kicking as well and see what I get

**Reply by PRProd** [reply]
@moh-Brent - I'm interested to hear about the Mi Box 3.  Does it actually run?  :smile: If so, does it get along with the Firemote card as-is?

**Comment by jrspowers** [comment]
Hallo @PRProd I can proudly say that the Mi Xiomi integration works very well, everything works, every app I have works immediately, the buttons work well, the volume works perfectly for me, may it be released to the original version!

**Reply by PRProd** [reply]
THANK YOU FOR YOUR TESTING!!

I'm glad to hear that everything is working for you!  Is your device model a "Mi Box S", or do you have something different?

**Reply by jrspowers** [reply]
Nee klop mi box S werk perfect !

**Reply by PRProd** [reply]
I tried translating **Nee klop** to English, but I think the meaning was lost.  It says "no knock" :laughing:

**Comment by PRProd** [comment]
# [Beta 2](https://github.com/PRProd/HA-Firemote/releases/tag/v2.1.0-beta2) is ready to download and test!!

## Changes:
 - Added additional SVG logos
 - Test fix for emby launch command (needs testing #18 @ChristophCaina)
 - Added Telenet TV shortcut (needs testing #56 @jrspowers)
 - Added Magenta TV shortcut (needs testing #58 @ChristophCaina)
 - Fixed a few SVG namespace collisions
- Enabled app launch access for TVNZ+ and Three Now on all devices (#55 @moh-Brent - needs testing)


## Not finished yet:
 - warning dialog/confirmation before executing a restart function

**Reply by ChristophCaina** [reply]
tested App-Shortcut on Nokia Streaming Box (runniing as shield TV Pro 2019)
- Jellyfin: OK
- Netfilx: OK
- Prime Video: OK
- Emby: failed
- YouTube: failed
- MagentaTV: failed

I will try to provide more information within tomorrow (or better within today^^)

**Reply by PRProd** [reply]
@ChristophCaina - YouTube failed, eh?  That's interesting.  Since Firemote doesn't officially support the Nokia Streaming Box yet, I won't be able to help you until I get going on #57.  As for the other two failures, I've addressed them on their respective threads #18 and #58.

**Comment by PRProd** [comment]
# [Beta 3](https://github.com/PRProd/HA-Firemote/releases/tag/v2.1.0-beta3) is ready to download and test

## Changes:
 - Default event changed for Mi Box S from event3 to event2.  This could have been the cause of the report from @moh-Brent in #55 that the "apps-button" was not doing anything
 - On the topic of the apps button not working, I replaced the Xiaomi command tied to that button with an adb start command which does not require a proper "event" number to be selected, and is faster than a "Strong" config.  Hopefully I've double solved this issue.
 - Adjusted the launch command for the emby app as reported by @ChristophCaina in #18 to be problematic.  This change should make it more stable and even faster for everyone who uses it.
 - Adjusted the launch command for Magenta TV as reported by @ChristophCaina in #58.  This needs testing/confirmation.
 - Added a confirmation dialog before executing a function: restart button press command as noted by @moh-Brent in #55 (good call Kiwi)

## Not finished yet:
 - nothing!  (let me know if I missed something)

**Reply by moh-Brent** [reply]
@PRProd 
Default event changed for Mi Box S from event3 to event2. This could have been the cause of the report from @moh-Brent in https://github.com/PRProd/HA-Firemote/discussions/55 that the "apps-button" was not doing anything
Works

Added a confirmation dialog before executing a function: restart button press command as noted by @moh-Brent in https://github.com/PRProd/HA-Firemote/discussions/55 (good call Kiwi)
Works

All perfect for me :) thank you soo much!

**Comment by PRProd** [comment]
THANK YOU EVERYBODY FOR YOUR TESTING!!!

I've officially launched [the new version](https://github.com/PRProd/HA-Firemote/releases/tag/v2.1.0)!  

If you have anything else you want to talk about regarding this new release, join the new conversation over at #60.
